packages/commerce-sdk-react/src/hooks/ShopperLogin/helper.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {ShopperLoginTypes} from 'commerce-sdk-isomorphic'
+import {ActionResponse, Argument} from '../types'
+import {useAsyncCallback} from '../useAsync'
+import useAuth from '../useAuth'
+import Auth from '../../auth'
+
+export enum ShopperLoginHelpers {
+    LoginGuestUser = 'loginGuestUser',
+    LoginRegisteredUserB2C = 'loginRegisteredUserB2C',
+    Logout = 'logout'
+}
+
+/**
+ * A hook for Public Client Shopper Login OAuth helpers.
+ * The hook calls the SLAS helpers imported from commerce-sdk-isomorphic.
+ * For more, see https://github.com/SalesforceCommerceCloud/commerce-sdk-isomorphic/#public-client-shopper-login-helpers
+ *
+ * Avaliable helpers:
+ * - loginRegisteredUserB2C
+ * - loginGuestUser
+ * - logout
+ */
+export function useShopperLoginHelper<Action extends ShopperLoginHelpers.LoginRegisteredUserB2C>(
+    action: Action
+): ActionResponse<Parameters<Auth['loginRegisteredUserB2C']>, ShopperLoginTypes.TokenResponse>
+export function useShopperLoginHelper<Action extends ShopperLoginHelpers.LoginGuestUser>(
+    action: Action
+): ActionResponse<never, ShopperLoginTypes.TokenResponse>
+export function useShopperLoginHelper<Action extends ShopperLoginHelpers.Logout>(
+    action: Action
+): ActionResponse<never, ShopperLoginTypes.TokenResponse>
+export function useShopperLoginHelper<Action extends ShopperLoginHelpers>(
+    action: Action
+): ActionResponse<any[], ShopperLoginTypes.TokenResponse> {
```

Comment:
Minor - about the generic `Action` type: I know that this pattern is also used in the stubs of other action hooks. But I feel there's a simpler way to define the types. 

```suggestion
export function useShopperLoginHelper(
    action: ShopperLoginHelpers.LoginRegisteredUserB2C
): ActionResponse<Parameters<Auth['loginRegisteredUserB2C']>, ShopperLoginTypes.TokenResponse>
export function useShopperLoginHelper(
    action: ShopperLoginHelpers.LoginGuestUser
): ActionResponse<never, ShopperLoginTypes.TokenResponse>
export function useShopperLoginHelper(
    action: ShopperLoginHelpers.Logout
): ActionResponse<never, ShopperLoginTypes.TokenResponse>
export function useShopperLoginHelper(
    action: ShopperLoginHelpers
): ActionResponse<
    Parameters<Auth['loginRegisteredUserB2C']> | never,
    ShopperLoginTypes.TokenResponse
> {
```

I'll leave it up to you. The above suggestion is also my practice in Typescript.